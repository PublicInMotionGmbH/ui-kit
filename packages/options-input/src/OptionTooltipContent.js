import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

const propTypes = {
  /** Title to show inside */
  title: PropTypes.string,

  /** Description to show */
  description: PropTypes.string
}

export const moduleName = 'options-input-tooltip'

/**
 * * Component which represents tooltip content for option.
 *
 * @param {object} props
 * @param {string} props.title
 * @param {string} props.description
 *
 * @returns {React.Element}
 */
function OptionTooltipContent (props) {
  const { title, description } = props

  const titleElement = (
    <span className={buildClassName([ moduleName, 'title' ])}>
      {title}
    </span>
  )

  const descriptionElement = description ? (
    <span className={buildClassName([ moduleName, 'description' ])}>
      {description}
    </span>
  ) : null

  return (
    <React.Fragment>
      {titleElement}
      {descriptionElement}
    </React.Fragment>
  )
}

OptionTooltipContent.propTypes = propTypes

export default OptionTooltipContent
