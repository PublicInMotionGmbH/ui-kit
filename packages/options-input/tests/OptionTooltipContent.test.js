import React from 'react'
import { shallow } from 'enzyme'

import OptionTooltipContent from '../src/OptionTooltipContent'

// Option for testing
const option = {
  id: 'person',
  icon: 'person',
  label: 'Adults',
  default: 1
}

const advancedOption = {
  id: 'person',
  icon: 'person',
  label: 'Adults',
  description: 'Older than 15',
  default: 1
}

describe('<OptionTooltipContent />', () => {
  it('renders children correctly', () => {
    const wrapper = shallow(<OptionTooltipContent title={option.label} description={option.description} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders advanced children correctly', () => {
    const wrapper = shallow(<OptionTooltipContent title={advancedOption.label} description={advancedOption.description} />)
    expect(wrapper).toMatchSnapshot()
  })
})
