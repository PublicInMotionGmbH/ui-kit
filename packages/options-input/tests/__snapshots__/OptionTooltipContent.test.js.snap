// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionTooltipContent /> renders advanced children correctly 1`] = `
<React.Fragment>
  <span
    className="talixo-options-input-tooltip__title"
  >
    Adults
  </span>
  <span
    className="talixo-options-input-tooltip__description"
  >
    Older than 15
  </span>
</React.Fragment>
`;

exports[`<OptionTooltipContent /> renders children correctly 1`] = `
<React.Fragment>
  <span
    className="talixo-options-input-tooltip__title"
  >
    Adults
  </span>
</React.Fragment>
`;
